wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Notification.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Simple notification overlay.
+ *
+ * @author  Alexander Ebert
+ * @copyright  2001-2019 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  Ui/Notification (alias)
+ * @module  WoltLabSuite/Core/Ui/Notification
+ */
+
+import * as Language from '../Language';
+
+type Callback = () => void;
+
+let _busy = false;
+let _callback: Callback | null = null;
+let _message: HTMLElement;
+let _notificationElement: HTMLElement | null = null;
+let _timeout: number;
+
+function init() {
+  if (_notificationElement === null) {
+    _notificationElement = document.createElement('div');
+    _notificationElement.id = 'systemNotification';
+
+    _message = document.createElement('p');
+    _message.addEventListener('click', hide);
+    _notificationElement.appendChild(_message);
+
+    document.body.appendChild(_notificationElement);
+  }
+}
+
+/**
+ * Hides the notification and invokes the callback if provided.
+ */
+function hide() {
+  clearTimeout(_timeout);
+
+  _notificationElement!.classList.remove('active');
```

Comment:
I'm not terribly happy with the need for the exclamation mark. Perhaps the element should be created when loading the module and the only thing `init()` does is attach it to the body?